src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/insert_specific_iterations.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+LinearIR::container InsertSpecificIterations::copy_loop(const LinearIR& linear_ir, const size_t loop_id) {
```

Comment:
I'm afraid of possible conflicts with the my PR https://github.com/openvinotoolkit/openvino/pull/21934

The main problem in this method is that we make a copy of `LinearIR` part, updates meta info `LoopInfo` in `LoopManager` of `LinearIR` and return this copy.
Thus, you have `LinearIR` with not inserted new expressions but `LoopManager` has been updated
What will do user with this copy - only god knows 😄 For example, he might insert the copy to another `LinearIR` - it means the original `LinearIR` will have `LoopInfos` with not existing ports.

I'd suggest to create another method `insert_copy_loop` as in the [PR](https://github.com/openvinotoolkit/openvino/pull/21934). What do you think?

Probably we should to revise the possibilities of copying of `LinearIR` parts in general.. Just thinking out loud

Reply:
Agree with your points

> I'd suggest to create another method insert_copy_loop as in the https://github.com/openvinotoolkit/openvino/pull/21934. What do you think?

I will take `insert_copy_loop` method and adopt it in my PR

Reply:
Done, please double check